specification/sql/resource-manager/Microsoft.Sql/preview/2017-10-01-preview/InterfaceEndpointProfiles.json

```diff
@@ -0,0 +1,416 @@
+          },
+          "202": {
+            "description": "Accepted"
+          },
+          "204": {
+            "description": "The specified interface endpoint profile does not exist."
+          }
+        },
+        "x-ms-long-running-operation": true,
+        "x-ms-examples": {
+          "Deletes interface endpoint profile.": {
+            "$ref": "./examples/InterfaceEndpointProfileDelete.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Sql/servers/{serverName}/interfaceEndpointProfiles": {
+      "get": {
+        "tags": [
+          "InterfaceEndpointProfiles"
+        ],
+        "description": "Gets a list of interface endpoint profiles attached to a server.",
+        "operationId": "InterfaceEndpointProfiles_ListByServer",
+        "parameters": [
+          {
+            "$ref": "#/parameters/ResourceGroupParameter"
+          },
+          {
+            "$ref": "#/parameters/ServerNameParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Succesfully retrieved the list of interface endpoint profiles.",
+            "schema": {
+              "$ref": "#/definitions/InterfaceEndpointProfilesListResult"
+            }
+          },
+          "default": {
+            "description": "*** Error Responses: ***\n\n * 404 SubscriptionDoesNotHaveServer - The requested server was not found\n\n * 404 ResourceNotFound - The requested resource was not found."
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        },
+        "x-ms-examples": {
+          "List of interface endpoint profiles on server.": {
+            "$ref": "./examples/InterfaceEndpointProfileListByServer.json"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "InterfaceEndpointProfileProperties": {
+      "description": "The properties of a interface endpoint profile.",
+      "required": [
+        "virtualNetworkSubnetId"
+      ],
+      "type": "object",
+      "properties": {
+        "virtualNetworkSubnetId": {
+          "description": "The ARM resource id of the virtual network subnet.",
+          "type": "string"
+        },
+        "privateIp": {
+          "description": "The Private ip associated with the interface endpoint profile",
+          "type": "string",
+          "readOnly": true
+        },
+        "state": {
```

Comment:
how is this property different from provisioningState? Are these the states the resource goes into during provisioning?

Reply:
This gives state to user based on sql metadata in the portal.